src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -844,6 +856,45 @@ protected BoundExpression BindInferredVariableInitializer(DiagnosticBag diagnost
             return expression;
         }
 
+        internal BoundExpression BindToNaturalType(BoundExpression expression, DiagnosticBag diagnostics)
+        {
+            BoundExpression result;
+            switch (expression)
+            {
+                case BoundConvertedSwitchExpression _:
+                case BoundConvertedTupleLiteral _:
+                    result = expression;
+                    break;
```

Comment:
>break [](start = 20, length = 5)

can we reach this? (seems like we should try to lock in the natural type only once)

Reply:
Yes, we can currently reach this.  It is harmless to lock in the natural type more than once.  It is a no-op if the expression has had its natural type locked in.

If you think there would be value in trying to ensure it happens only once, please file an issue.  We can use the WasConverted flag to detect duplicate application of binding to the natural type.

---
In reply to: [301783223](https://github.com/dotnet/roslyn/pull/37052#discussion_r301783223) [](ancestors = 301783223)